src/main/kotlin/ecommerce/repository/ProductRepository.kt

```diff
@@ -0,0 +1,51 @@
+package ecommerce.repository
+
+import ecommerce.model.Product
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.core.RowMapper
+import org.springframework.stereotype.Repository
+import java.sql.ResultSet
+
+@Repository
+class ProductRepository(private val jdbcTemplate: JdbcTemplate) {
```

Comment:
We've seen `@Controller` and now here's `@Repository`.
What do you think are their similarities and differences?